performance_tests/src/aws_encryption_sdk_performance_tests/keyrings/hierarchy_keyring.py

```diff
@@ -0,0 +1,174 @@
+        encryption_context: Dict[str, str] = param.encryption_context
+
+        if b"tenant" not in encryption_context:
+            raise ValueError("EncryptionContext invalid, does not contain expected tenant key value pair.")
+
+        tenant_key_id: str = encryption_context.get(b"tenant")
+        branch_key_id: str
+
+        if tenant_key_id == b"TenantA":
+            branch_key_id = self.branch_key_id_for_tenant_a
+        elif tenant_key_id == b"TenantB":
+            branch_key_id = self.branch_key_id_for_tenant_b
+        else:
+            raise ValueError(f"Item does not contain valid tenant ID: {tenant_key_id=}")
+
+        return GetBranchKeyIdOutput(branch_key_id=branch_key_id)
+
+
+def create_keyring(
+    key_store_table_name: str,
+    logical_key_store_name: str,
+    kms_key_id: str
+):
+    """Demonstrate how to create a hierarchy keyring.
+
+    Usage: create_keyring(key_store_table_name, logical_key_store_name, kms_key_id)
+    :param key_store_table_name: Name of the KeyStore DynamoDB table.
+    :type key_store_table_name: string
+    :param logical_key_store_name: Logical name of the KeyStore.
+    :type logical_key_store_name: string
+    :param kms_key_id: KMS Key identifier for the KMS key you want to use.
+    :type kms_key_id: string
+
+    For more information on KMS Key identifiers, see
+    https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id
+    """
+    # Create boto3 clients for DynamoDB and KMS.
+    ddb_client = boto3.client('dynamodb', region_name="us-west-2")
+    kms_client = boto3.client('kms', region_name="us-west-2")
+
+    # Configure your KeyStore resource.
+    # This SHOULD be the same configuration that you used
+    # to initially create and populate your KeyStore.
+    keystore: KeyStore = KeyStore(
+        config=KeyStoreConfig(
+            ddb_client=ddb_client,
+            ddb_table_name=key_store_table_name,
+            logical_key_store_name=logical_key_store_name,
+            kms_client=kms_client,
+            kms_configuration=KMSConfigurationKmsKeyArn(
+                value=kms_key_id
+            ),
+        )
+    )
+
+    # Call CreateKey to create two new active branch keys
+    branch_key_id_a: str = PerfTestUtils.DEFAULT_BRANCH_KEY_ID_A
+    branch_key_id_b: str = PerfTestUtils.DEFAULT_BRANCH_KEY_ID_B
+
+    # Create a branch key supplier that maps the branch key id to a more readable format
+    branch_key_id_supplier: IBranchKeyIdSupplier = ExampleBranchKeyIdSupplier(
+        tenant_1_id=branch_key_id_a,
+        tenant_2_id=branch_key_id_b,
+    )
+
+    # Create the Hierarchical Keyring.
+    mat_prov: AwsCryptographicMaterialProviders = AwsCryptographicMaterialProviders(
+        config=MaterialProvidersConfig()
+    )
+
+    keyring_input: CreateAwsKmsHierarchicalKeyringInput = CreateAwsKmsHierarchicalKeyringInput(
+        key_store=keystore,
+        branch_key_id_supplier=branch_key_id_supplier,
```

Comment:
Let's just one one branch key ID. A supplier is overkill for a performance test
```suggestion
        branch_key_id=...
```

Reply:
Done